FV1000Reader: fix "---" filter models

stelfrich opened the pull request:
Under some circumstances, filter names in OIBs/OIFs are `"---"` instead of `""`. This triggers parsing of the transmittance range cut-in and cut-out values from the name which raises an ArrayIndexOutOfBoundsException in [FV1000Reader.java#L1103](https://github.com/openmicroscopy/bioformats/blob/9088cd5a056d742066d2befdf19a83c71ec76e5d/components/formats-gpl/src/loci/formats/in/FV1000Reader.java#L1103).

sbesson commented:
Hi @stelfrich, thanks for the Pull Request. Two initial comments:

- would you be able to share a sample file reproducing the issue that we could add to our non-regression test suite?
- after running the automated tests against our OIB/OIF samples with this PR included, the changes introduced in 9b92021 cause a NPE to be thrown (see [log](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-repository-subset/429)) under some conditions. The `---` handling might require an extra non-null check and could probably be moved alongside the other checks in https://github.com/openmicroscopy/bioformats/blob/v5.5.1/components/formats-gpl/src/loci/formats/in/FV1000Reader.java#L534.

stelfrich commented:
> would you be able to share a sample file reproducing the issue that we could add to our non-regression test suite?

http://qa.openmicroscopy.org.uk/qa/feedback/17748/

> The `---` handling might require an extra non-null check and could probably be moved alongside the other checks in https://github.com/openmicroscopy/bioformats/blob/v5.5.1/components/formats-gpl/src/loci/formats/in/FV1000Reader.java#L534.

That's perfectly right. I have moved the checks to that location in 1ac08ae. Since I wasn't sure about your force-pushing policies on "feature" branches, I have left the initial commit untouched and introduced one that reverts it. If test are passing, we can remove the first two commits of this PR to not pollute the history..

sbesson commented:
The latest [respository subset job](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-repository-subset/431/) passed with the latest changes. Including this PR tomorrow's build for code review and we will include your sample file in our repository. Feel free to force-push the commits to the latest state of this branch.

rleigh-codelibre commented:
Looking over the code changes, this looks fine and good to merge.

sbesson commented:
https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-full-repository/557/ is now green including the new sample files. While reviewing the code, I realized the whole channel metadata population section could use some refactoring including some usage of the `DataTools.parseDouble` methods but I'll open a follow-up PR for that.

Thanks for the contribution @stelfrich, merging.
